pkg/controller/atlasinventory/atlasinventory_controller.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+// MongoDBAtlasInventoryReconciler reconciles a MongoDBAtlasInventory object
+type MongoDBAtlasInventoryReconciler struct {
+	Client client.Client
+	watch.ResourceWatcher
+	Log             *zap.SugaredLogger
+	Scheme          *runtime.Scheme
+	AtlasDomain     string
+	GlobalAPISecret client.ObjectKey
+	EventRecorder   record.EventRecorder
+}
+
+// Dev note: duplicate the permissions in both sections below to generate both Role and ClusterRoles
+
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=MongoDBAtlasInventorys,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=atlas.mongodb.com,resources=MongoDBAtlasInventorys/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+
+// +kubebuilder:rbac:groups=atlas.mongodb.com,namespace=default,resources=MongoDBAtlasInventorys,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=atlas.mongodb.com,namespace=default,resources=MongoDBAtlasInventorys/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",namespace=default,resources=secrets,verbs=get;list;watch
+
+func (r *MongoDBAtlasInventoryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = ctx
+	log := r.Log.With("MongoDBAtlasInventory", req.NamespacedName)
+
+	inventory := &dbaas.MongoDBAtlasInventory{}
+	if err := r.Client.Get(ctx, req.NamespacedName, inventory); err != nil {
+		if errors.IsNotFound(err) {
+			// CR deleted since request queued, child objects getting GC'd, no requeue
+			log.Info("DBaaSInventory resource not found, has been deleted")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Error fetching MongoDBAtlasInventory for reconcile")
+		return ctrl.Result{}, err
+	}
+
+	if inventory.ConnectionSecretObjectKey() != nil {
+		// Note, that we are not watching the global connection secret - seems there is no point in reconciling all
+		// the services once that secret is changed
+		r.EnsureResourcesAreWatched(req.NamespacedName, "Secret", log, *inventory.ConnectionSecretObjectKey())
+	}
+
+	log.Infow("-> Starting MongoDBAtlasInventory reconciliation", "spec", inventory.Spec)
+
+	// This update will make sure the status is always updated in case of any errors or successful result
+	defer func(inv *dbaas.MongoDBAtlasInventory) {
+		err := r.Client.Status().Update(ctx, inv)
+		if err != nil {
+			log.Infow("Could not update resource status:%v", err)
+		}
+	}(inventory)
+
+	inventory.Status.Type = dbaas.ProviderName
+	atlasConn, err := atlas.ReadConnection(log, r.Client, r.GlobalAPISecret, inventory.ConnectionSecretObjectKey())
+	if err != nil {
+		result := workflow.Terminate(workflow.AtlasCredentialsNotProvided, err.Error())
+		dbaas.SetInventoryCondition(inventory, string(status.ReadyType), metav1.ConditionFalse, string(result.Reason()), result.Message())
```

Comment:
I believe the condition can be set according to mentioned in the API design document-> 

<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-9ecc739f-7fff-ccac-898d-49a7fded9fa4"><div dir="ltr" style="margin-left:0pt;" align="left">
Type | Status | Reason
-- | -- | --
SpecSynced | True | SyncOK
SpecSynced | False | InputError
SpecSynced | False | BackendError
SpecSynced | False | EndpointUnreachable
SpecSynced | False | AuthenticationError

</div></b>

Reply:
OK. PR updated.

Reply:
Thanks, quick question why we need this extra condition   `ReadyForBinding`
is that possible can we use existing condition `SpecSynced`

Reply:
oh, I see please ignore -This is required for connection cr


<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-9fb40fb9-7fff-c762-7e1c-fa10f884a788"><div dir="ltr" style="margin-left:0pt;" align="left">
Type | Status | Reason
-- | -- | --
ReadyForBinding | True | Ready
ReadyForBinding | False | Unreachable
ReadyForBinding | False | NotFound
ReadyForBinding | False | BackendError
ReadyForBinding | False | AuthenticationError

</div></b>

Reply:
@redhatHameed this is based on the API design ADR.